packages/core/core-flows/src/product/workflows/create-product-variants.ts

```diff
@@ -50,6 +54,41 @@ const buildLink = (
   return link
 }
 
+const validateVariantsDuplicateInventoryItemIds = (
+  variantsData: {
+    variantId: string
+    inventory_items: {
+      inventory_item_id: string
+      required_quantity?: number
+    }[]
+  }[]
+) => {
+  const errors: string[] = []
+
+  for (const variantData of variantsData) {
+    const inventoryItemIds = variantData.inventory_items.map(
+      (item) => item.inventory_item_id
+    )
+    const duplicatedInventoryItemIds = inventoryItemIds.filter(
+      (id, index) => inventoryItemIds.indexOf(id) !== index
+    )
+
+    if (duplicatedInventoryItemIds.length) {
+      errors.push(
+        `Variant with id ${
+          variantData.variantId
+        } has duplicate inventory item ids ${duplicatedInventoryItemIds.join(
+          ", "
```

Comment:
suggestion(non-blocking): Simplify error message. Don't think the IDs help here, so if this bubbles up to admin, then it would be a more readable message
```suggestion
        `Cannot associate duplicate inventory items to a variant`
```

Reply:
Also, would maybe argue we can just throw on the first error we find, but I'll let you decide.

Reply:
The idea is the following, since the api can be consumed outside of the admin, if you send multiple variants with duplicated inventory items, you get the list of all variants to change instead of receiving one, change it, send them back and get another error. If thats ok for you ill keep it, but ill include your previous comment anyway

Reply:
Yeah, I am aware why we are usually doing this. Fine to keep it as is 👍